MapboxNavigation/MapboxSpeechController.swift

```diff
@@ -0,0 +1,157 @@
+import Foundation
+import CoreLocation
+import AVFoundation
+import MapboxSpeech
+import MapboxCoreNavigation
+import MapboxDirections
+
+/**
+ `MapboxVoiceController` extends the default `RouteVoiceController` by providing a more robust speech synthesizer via the Mapbox Speech API. `RouteVoiceController` will be used as a fallback during poor network conditions.
+ */
+@objc(MBMapboxVoiceController)
+public class MapboxVoiceController: RouteVoiceController {
+    
+    /**
+     Number of seconds a request can wait before it is canceled and the default speech synthesizer speaks the instruction.
+     */
+    @objc public var timeoutIntervalForRequest:TimeInterval = 5
+    
+    /**
+     Number of steps ahead of the current step to cache spoken instructions.
+     */
+    @objc public var stepsAheadToCache: Int = 3
+    
+    var audioTask: URLSessionDataTask?
+    var spokenInstructionsForRoute = NSCache<NSString, NSData>()
+    
+    var speech: SpeechSynthesizer
+    var locale: Locale
+    
+    let localizedErrorMessage = NSLocalizedString("FAILED_INSTRUCTION", bundle: .mapboxNavigation, value: "Unable to read instruction aloud.", comment: "Error message when the SDK is unable to read a spoken instruction.")
+    
+    public init(locale: Locale, accessToken: String? = nil, host: String? = nil) {
```

Comment:
I don’t think `locale` should be an argument on the initializer. A developer using just MapboxSpeechController in conjunction with RouteController (but not NavigationViewController) would need to manually set the locale, which runs counter to the goal of making localization as transparent as possible.

`didPassSpokenInstructionPoint(_:)` has access to the RouteProgress:

https://github.com/mapbox/mapbox-navigation-ios/blob/f12ea1b8ecb42c127098d9eea7647e233d9ee801/MapboxNavigation/MapboxSpeechController.swift#L50

so RouteOptions is also easily accessible:

```swift
let routeProgresss = notification.userInfo![MBRouteControllerDidPassSpokenInstructionPointRouteProgressKey] as! RouteProgress
let routeOptions = routeProgress.route.routeOptions
…
cacheSpokenInstruction(instruction: instruction, locale: routeOptions.locale)
```

Streamlining localization is also relevant to RouteVoiceController’s AVSpeechSynthesizer fallback, so I opened #933 to track the general issue.